[ao][sparsity] Support for the nn.Parameter in BaseDataSparsifier

macandro96 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #79535
* #79534
* #79254
* __->__ #79253
* #79427
* #79252
* #79251

Users can now just pass in a nn.Parameter (or layer.weight) to the Data Sparsifier.
Note: The data sparsifier stores the passed nn.Parameter as a new parameter in the internal container module whose requires_grad=False.
So, essentialy when the parameter is trained, it's new values are not reflected inside the data sparsifier class

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataSparsifier```

Differential Revision: [D37164116](https://our.internmc.facebook.com/intern/diff/D37164116)

Commit:
[ao][sparsity] Support for the nn.Parameter in BaseDataSparsifier

Users can now just pass in a nn.Parameter (or layer.weight) to the Data Sparsifier.
Note: The data sparsifier stores the passed nn.Parameter as a new parameter in the internal container module whose requires_grad=False.
So, essentialy when the parameter is trained, it's new values are not reflected inside the data sparsifier class

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataSparsifier```

[ghstack-poisoned]

HDCharles commented on test/ao/sparsity/test_data_sparsifier.py:
```diff
@@ -25,65 +26,91 @@ class TestBaseDataSparsifier(TestCase):
     def test_constructor(self):
```
its probably a good idea to start splitting these tests a bit so they don't continue to grow into a supertest.

I'd recommend creating a helper class to run the test which takes as an input a tuple of tensors/params and then does the checks

then have the main tests be test_constructor_parameters and test_constructor_tensors which just passes in the set of tensors or parameters. Otherwise, if the unit tests start to fail it becomes hard to pinpoint the reason, is it because of the parameter handling? tensor? ...etc

macandro96 replied:
That's a good point - let me try doing that.

macandro96 replied:
I've revamped the unit-tests - please let me know if it looks good

Commit:
Update on "[ao][sparsity] Support for the nn.Parameter in BaseDataSparsifier"


Users can now just pass in a nn.Parameter (or layer.weight) to the Data Sparsifier.
Note: The data sparsifier stores the passed nn.Parameter as a new parameter in the internal container module whose requires_grad=False.
So, essentialy when the parameter is trained, it's new values are not reflected inside the data sparsifier class

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataSparsifier```

[ghstack-poisoned]

Commit:
Update on "[ao][sparsity] Support for the nn.Parameter in BaseDataSparsifier"


Users can now just pass in a nn.Parameter (or layer.weight) to the Data Sparsifier.
Note: The data sparsifier stores the passed nn.Parameter as a new parameter in the internal container module whose requires_grad=False.
So, essentialy when the parameter is trained, it's new values are not reflected inside the data sparsifier class

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataSparsifier```

[ghstack-poisoned]

Commit:
Update on "[ao][sparsity] Support for the nn.Parameter in BaseDataSparsifier"


Users can now just pass in a nn.Parameter (or layer.weight) to the Data Sparsifier.
Note: The data sparsifier stores the passed nn.Parameter as a new parameter in the internal container module whose requires_grad=False.
So, essentialy when the parameter is trained, it's new values are not reflected inside the data sparsifier class

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataSparsifier```

[ghstack-poisoned]

Commit:
Update on "[ao][sparsity] Support for the nn.Parameter in BaseDataSparsifier"


Users can now just pass in a nn.Parameter (or layer.weight) to the Data Sparsifier.
Note: The data sparsifier stores the passed nn.Parameter as a new parameter in the internal container module whose requires_grad=False.
So, essentialy when the parameter is trained, it's new values are not reflected inside the data sparsifier class

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataSparsifier```

[ghstack-poisoned]

Commit:
Update on "[ao][sparsity] Support for the nn.Parameter in BaseDataSparsifier"


Users can now just pass in a nn.Parameter (or layer.weight) to the Data Sparsifier.
Note: The data sparsifier stores the passed nn.Parameter as a new parameter in the internal container module whose requires_grad=False.
So, essentialy when the parameter is trained, it's new values are not reflected inside the data sparsifier class

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataSparsifier```

[ghstack-poisoned]

z-a-f reviewed (approved):
lgtm

Commit:
Update on "[ao][sparsity] Support for the nn.Parameter in BaseDataSparsifier"


Users can now just pass in a nn.Parameter (or layer.weight) to the Data Sparsifier.
Note: The data sparsifier stores the passed nn.Parameter as a new parameter in the internal container module whose requires_grad=False.
So, essentialy when the parameter is trained, it's new values are not reflected inside the data sparsifier class

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataSparsifier```

[ghstack-poisoned]

Commit:
Update on "[ao][sparsity] Support for the nn.Parameter in BaseDataSparsifier"


Users can now just pass in a nn.Parameter (or layer.weight) to the Data Sparsifier.
Note: The data sparsifier stores the passed nn.Parameter as a new parameter in the internal container module whose requires_grad=False.
So, essentialy when the parameter is trained, it's new values are not reflected inside the data sparsifier class

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataSparsifier```

[ghstack-poisoned]

Commit:
Update on "[ao][sparsity] Support for the nn.Parameter in BaseDataSparsifier"


Users can now just pass in a nn.Parameter (or layer.weight) to the Data Sparsifier.
Note: The data sparsifier stores the passed nn.Parameter as a new parameter in the internal container module whose requires_grad=False.
So, essentialy when the parameter is trained, it's new values are not reflected inside the data sparsifier class

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataSparsifier```

[ghstack-poisoned]

macandro96 commented:
@macandro96 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37164116).

Commit:
Update on "[ao][sparsity] Support for the nn.Parameter in BaseDataSparsifier"


Users can now just pass in a nn.Parameter (or layer.weight) to the Data Sparsifier.
Note: The data sparsifier stores the passed nn.Parameter as a new parameter in the internal container module whose requires_grad=False.
So, essentialy when the parameter is trained, it's new values are not reflected inside the data sparsifier class

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataSparsifier```

Differential Revision: [D37164116](https://our.internmc.facebook.com/intern/diff/D37164116)

[ghstack-poisoned]

macandro96 commented:
@macandro96 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37164116).

Commit:
Update on "[ao][sparsity] Support for the nn.Parameter in BaseDataSparsifier"


Users can now just pass in a nn.Parameter (or layer.weight) to the Data Sparsifier.
Note: The data sparsifier stores the passed nn.Parameter as a new parameter in the internal container module whose requires_grad=False.
So, essentialy when the parameter is trained, it's new values are not reflected inside the data sparsifier class

Test Plan:
```python test/test_ao_sparsity.py TestBaseDataSparsifier```

Differential Revision: [D37164116](https://our.internmc.facebook.com/intern/diff/D37164116)

[ghstack-poisoned]
